test/external/dataplane/connectivity.go

```diff
@@ -40,165 +40,179 @@ var (
-		f.CreateServiceExport(np.Config.Cluster, svc.Name)
+	By(fmt.Sprintf("Creating a pod and a service in cluster %q", clusterName))
 
-		// Get globalIPs for the network pod to use later
-		remoteIP := f.AwaitGlobalIngressIP(np.Config.Cluster, svc.Name, svc.Namespace)
-		Expect(remoteIP).ToNot(Equal(""))
+	np := p.Framework.NewNetworkPod(&framework.NetworkPodConfig{
+		Type:          framework.CustomPod,
+		Port:          80,
+		Cluster:       p.Cluster,
+		Scheduling:    p.ClusterScheduling,
+		Networking:    p.Networking,
+		ContainerName: testContainerName,
+		ImageName:     testImage,
+		Command:       simpleHTTPServerCommand,
+	})
+	svc := np.CreateService()
 
-		podGlobalIPs := f.AwaitClusterGlobalEgressIPs(np.Config.Cluster, constants.ClusterGlobalEgressIPName)
-		Expect(len(podGlobalIPs)).ToNot(BeZero())
-		podGlobalIP := podGlobalIPs[0]
+	// Get handle for existing docker
+	docker := framework.New(extAppName)
 
-		By(fmt.Sprintf("Sending an http request from external app %q to the service %q in the cluster %q",
-			dockerIP, remoteIP, clusterName))
+	// Get IPs to use later
+	podIP := np.Pod.Status.PodIP
+	svcIP := svc.Spec.ClusterIP
+	dockerIP := docker.GetIP(extNetName)
 
-		command := []string{"curl", "-m", "10", fmt.Sprintf("%s:%d/%s%s", remoteIP, 80, f.Namespace, clusterName)}
-		_, _ = docker.RunCommand(command...)
+	var targetIP string
 
-		By("Verifying the pod received the request")
+	switch p.ToEndpointType {
+	case tcp.PodIP:
+		targetIP = podIP
+	case tcp.ServiceIP:
+		targetIP = svcIP
+	default:
+		framework.Failf("Unsupported ToEndpointType %v was passed", p.ToEndpointType)
+	}
 
-		podLog := np.GetLog()
-		if framework.ClusterIndex(idx) == extClusterIdx {
-			// TODO: current behavior is that source IP from external app to the pod in the cluster that directly connected to
-			// external network is the gateway IP of the pod network. Consider if it can be consistent.
-			Expect(podLog).To(MatchRegexp(".*GET /%s%s .*", f.Namespace, clusterName))
-		} else {
-			Expect(podLog).To(MatchRegexp("%s .*GET /%s%s .*", extEgressGlobalIP, f.Namespace, clusterName))
-		}
+	By(fmt.Sprintf("Sending an http request from external app %q to %q in the cluster %q",
+		dockerIP, targetIP, clusterName))
 
-		framework.Logf("%s", podLog)
+	command := []string{"curl", "-m", "10", fmt.Sprintf("%s:%d/%s%s", targetIP, 80, p.Framework.Namespace, clusterName)}
+	_, _ = docker.RunCommand(command...)
 
-		By(fmt.Sprintf("Sending an http request from the test pod %q %q in cluster %q to the external app's ingressGlobalIP %q",
-			np.Pod.Name, podGlobalIP, clusterName, extIngressGlobalIP))
+	By("Verifying the pod received the request")
 
-		cmd := []string{"curl", "-m", "10", fmt.Sprintf("%s:%d/%s%s", extIngressGlobalIP, 80, f.Namespace, clusterName)}
-		_, _ = np.RunCommand(cmd)
+	podLog := np.GetLog()
 
-		By(fmt.Sprintf("Verifying that external app received request from egressGlobalIP %q", podGlobalIP))
+	if clusterName == externalClusterName {
```

Comment:
Can you please add some comments on why the validation is different if the clusterName matches with externalClusterName.

Reply:
Added an ascii art of the test environment and expected behaviors.

Reply:
Thank you